### DISCUSSION TITLE
For getting the Labels of fields while parsing EDI document

### DISCUSSION THREAD

**Discussion by harshilKapasi** [xlate/staedi#491] [Q&A]
I am using the library for parsing from EDI to json object. Here, I want to add the labels to the EDI fields here, and i got to know that it can be done using the schemas, but i am not able to find any schema that can be used as a standard schema, like, for the fields of UNB, UNE, etc.
where can i get these schemas? (so that my output becomes more readable)
I got across some schemas online, but they give error on using in the library, like below 
`Caused by: io.xlate.edi.internal.schema.StaEDISchemaReadException: Unexpected XML element [{http://www.w3.org/2001/XMLSchema}schema]`

can anyone help here?

It will also be great if you can provide schema for these which will work for Staedi

**Comment by MikeEdgar** [comment]
Hi @harshilKapasi , which schema is causing that exception? You're looking for only the descriptions of the control segments (e.g. UNB/UNE)?

**Reply by harshilKapasi** [reply]
I want to generate json which has the element names as the keys instead of their codes, means for UNB, the final json should consist of keys with the corresponding field names as keys and their values, for that i have to get the actual schemas, which can give me the names for the fields, for example: UNG has field MessageGroupIdentification_1 which is of code E0038

can you help here?
Thanks
